ibis/backends/pyspark/tests/conftest.py

```diff
@@ -22,22 +25,41 @@ def set_pyspark_database(con, database):
 class TestConf(BackendTest):
     deps = ("pyspark",)
 
-    def _load_data(self, **_: Any) -> None:
+    def _load_data_helper(self, for_streaming: bool = False):
         from pyspark.sql import Row
 
+        get_table_name = (
+            (lambda name: f"{name}_streaming") if for_streaming else (lambda name: name)
+        )
+
         s = self.connection._session
         num_partitions = 4
-
         sort_cols = {"functional_alltypes": "id"}
 
+        if for_streaming:
+            # "By default, Structured Streaming from file based sources
+            # requires you to specify the schema, rather than rely on
+            # Spark to infer it automatically. ... .  For ad-hoc use
+            # cases, you can re-enable schema inference by setting
+            # spark.sql.streaming.schemaInference to true."
+            # Ref: https://spark.apache.org/docs/latest/structured-streaming-programming-guide.html#schema-inference-and-partition-of-streaming-dataframesdatasets
+            s.sql("set spark.sql.streaming.schemaInference=true")
```

Comment:
You can pass this as a config to the session builder https://github.com/ibis-project/ibis/blob/b8894d01fde9cba7dc06b2182b7ee5f42f6903d8/ibis/backends/pyspark/tests/conftest.py#L169-L191. I think that helps group everything together instead of having this single line of SQL here